alchemist-physics/src/main/kotlin/it/unibo/alchemist/model/implementations/reactions/PhysicsUpdate.kt

```diff
@@ -18,22 +18,26 @@ import it.unibo.alchemist.model.interfaces.Dependency
 import it.unibo.alchemist.model.interfaces.Environment
 import it.unibo.alchemist.model.interfaces.GlobalReaction
 import it.unibo.alchemist.model.interfaces.Time
+import it.unibo.alchemist.model.interfaces.TimeDistribution
 import it.unibo.alchemist.model.interfaces.environments.Dynamics2DEnvironment
 import org.danilopianini.util.ImmutableListSet
 import org.danilopianini.util.ListSet
 
 /**
  * A global Reaction responsible for updating the physics of an [Dynamics2DEnvironment].
  */
-class PhysicsUpdate<T> @JvmOverloads constructor(
+class PhysicsUpdate<T>(
     /**
      * The environment to update.
      */
     val environment: Dynamics2DEnvironment<T>,
-    updateRate: Double = 1.0,
+    override val timeDistribution: TimeDistribution<T>,
 ) : GlobalReaction<T> {
 
-    override val timeDistribution: DiracComb<T> = DiracComb(updateRate)
+    constructor(
+        environment: Dynamics2DEnvironment<T>,
+        updateRate: Double,
+    ) : this(environment, DiracComb(updateRate))
```

Comment:
what'd be a reasonable default? 30Hz sounds reasonable

```suggestion
    @JvmOverloads constructor(
        environment: Dynamics2DEnvironment<T>,
        updateRate: Double = 30.0,
    ) : this(environment, DiracComb(updateRate))
```

Reply:
I could add a default here, but then any time-distribution specified in the configuration will be ignored. Do we want this behavior?

Reply:
No we don't but I don't understand why that would be the behaviour. We would end up with three constructors:
* Environment+TimeDistribution
* Environment (+ default rate/timedistribution)
* Environment + rate

Why would a custom time distribution be ignored? It is the first constructor

Reply:
Okay, I was sure I tried this, and it wasn't working. 
Now I've tried it again and works as you'd expect.

Reply:
I tried --> I wrote a test for it

![image](https://user-images.githubusercontent.com/1991673/178751421-e45c4014-ba52-4b29-a4bc-cfb161b79589.png)

Reply:
Actually, I did write the test 😂 ("Customize rate with time-distribution")